File: BACK-END/src/main/java/org/kuenteco/backend/dto/logic/transaction/UserProfilesWithTransactionsDTO.java
package org.kuenteco.backend.dto.logic.transaction;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserProfilesWithTransactionsDTO {
    private String username;
    private String email;
    private List<ProfileWithTransactionsDTO> profiles;
    private Integer totalProfiles;
    private Integer totalTransactions;
}

File: FRONT-END/lib/dto/transaction/kuenteco/user_profiles_with_transactions_dto.dart
import 'profile_with_transactions_dto.dart';

class UserProfilesWithTransactionsDTO {
  final String username;
  final String email;
  final List<ProfileWithTransactionsDTO> profiles;
  final int totalProfiles;
  final int totalTransactions;

  UserProfilesWithTransactionsDTO({
    required this.username,
    required this.email,
    required this.profiles,
    required this.totalProfiles,
    required this.totalTransactions,
  });

  factory UserProfilesWithTransactionsDTO.fromJson(Map<String, dynamic> json) {
    return UserProfilesWithTransactionsDTO(
      username: json['username'],
      email: json['email'],
      profiles: (json['profiles'] as List)
          .map((item) => ProfileWithTransactionsDTO.fromJson(item))
          .toList(),
      totalProfiles: json['totalProfiles'],
      totalTransactions: json['totalTransactions'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'email': email,
      'profiles': profiles.map((item) => item.toJson()).toList(),
      'totalProfiles': totalProfiles,
      'totalTransactions': totalTransactions,
    };
  }
}
